autoload/go/job.vim

```diff
@@ -104,15 +104,20 @@ function go#job#Spawn(args)
   let cbs.close_cb = function('s:close_cb', [], state)
 
   function state.show_errors(job, exit_status, data)
+    let l:winid = win_getid(winnr())
+    call win_gotoid(self.winid)
+
     let l:listtype = go#list#Type(self.for)
     if a:exit_status == 0
       call go#list#Clean(l:listtype)
+      call win_gotoid(l:winid)
```

Comment:
This is slightly confusing. We jump two times.  First to the windows where the job was started, and then to the current window. Why do we have a logic like this? If we jump to the window the job was started, I wouldn't want to go back, but I think here we jump back because there are no errors right ? Maybe we should have an `if .. else ...` clause instead of jumping twice.

Reply:
Assume the user has 2 windows open. While in window 1, the user runs an async command whose list type is `locationlist`. Before the async command completes, the user moves to window 2. Then the async command completes successfully. We need to clean the list of window 1 (hence the first jump). And we want to avoid stealing window focus, so we make sure to make window 2 active again before returning from the callback.